src/vm/node_modules/VM.js

```diff
@@ -10957,6 +10958,271 @@ exports.receive = function (target, options, callback)
     });
 };
 
+/*
+ * Bhyve-specific reprovision function that replaces disk0 with a new image
+ * while preserving all other disks and configuration.
+ */
+function reprovisionBhyve(uuid, vmobj, payload, log, cb)
+{
+    var disk0;
+    var disk0_ds;
+    var disk0_path;
+    var snapshot;
+
+    assert(vmobj.brand === 'bhyve', 'reprovisionBhyve called on non-bhyve VM');
+    assert(payload.image_uuid, 'missing image_uuid in payload');
+
+    log.info('Starting bhyve reprovision for VM %s', uuid);
+
+    vasync.waterfall([
+        function findDisk0(next) {
+            // Find disk0 in the VM's disk list by looking for the one
+            // ending with /disk0
+            if (!vmobj.disks || vmobj.disks.length === 0) {
+                next(new Error('No disks found for bhyve VM'));
+                return;
+            }
+
+            var i;
+            for (i = 0; i < vmobj.disks.length; i++) {
+                if (vmobj.disks[i].path) {
+                    // Extract dataset name from path
+                    var ds = vmobj.disks[i].path
+                        .replace(/^\/dev\/zvol\/rdsk\//, '')
+                        .replace(/^\/dev\/zvol\/dsk\//, '');
+                    // Check if this is disk0
+                    if (ds.match(/\/disk0$/)) {
+                        // Found disk0, verify it has boot property set
+                        if (vmobj.disks[i].boot !== true) {
+                            next(new Error('disk0 found but does not have '
+                                + 'boot property set to true - reprovision '
+                                + 'requires a bootable disk0'));
+                            return;
```

Comment:
Nice way to deal with this corner-case.  :)  Someone who does this can probably figure out `zfs rename` if need be.

And per TritonDataCenter/illumos-joyent#518 -> SO sorry about `install` not being a good name for a make target in illumos.